pynicotine/gtkgui/frame.py

```diff
@@ -1468,14 +1468,45 @@ def on_update_user_info(self, *_args):
     def on_get_user_info(self, *_args):
         self.userinfo.on_get_user_info()
 
-    """ Browse Shares """
+    """ Shares """
 
     def on_get_shares(self, *_args):
         self.userbrowse.on_get_shares()
 
     def on_load_from_disk(self, *_args):
         self.userbrowse.on_load_from_disk()
 
+    def shares_unavailable_response(self, _dialog, response_id, _data):
+
+        if response_id == 2:  # 'Retry'
+            self.core.shares.rescan_shares()
+
+        elif response_id == 3:  # 'Force Rescan'
+            self.core.shares.rescan_shares(force=True)
+
+    def shares_unavailable(self, shares):
+
+        shares_list_message = ""
+
+        for virtual_name, folder_path in shares:
+            shares_list_message += "• %s (%s)\n" % (folder_path, virtual_name)
+
+        def create_dialog():
+            OptionDialog(
+                parent=self.window,
+                title=_("Shares Not Available"),
+                message=_("The following shares are not available. Verify that external disks are mounted and "
+                          "folder permissions are correct."),
```

Comment:
> strings are not necessarily final

"~_The following shares are not available._~ Verify that any external disks are correctly mounted and accessible."